ControlRoomApplication/ControlRoomApplication/GUI/MainForm.cs

```diff
@@ -643,75 +655,104 @@ private void groupBox3_Enter(object sender, EventArgs e)
         {
-            finalSettings = !finalSettings;
-            if (finalSettings == false)
-            {
+            // Begin User Validation of input IP and port numbers
+            bool MCUPortValid, MCUIPValid, PLCPortValid, WCOMPortValid;
+            MCUPortValid = Validator.ValidatePort(txtMcuCOMPort.Text);
+            PLCPortValid = Validator.ValidatePort(txtPLCPort.Text);
+            WCOMPortValid = Validator.ValidatePort(txtWSCOMPort.Text);
+            MCUIPValid = Validator.ValidateIPAddress(txtPLCIP.Text);
 
-                //Editing text to relflect state -- When the settings are being edited, the button will need to say 'finalize'
-                acceptSettings.Text = "Edit Settings";
-                if (comboBox2.Text != "Production Weather Station")
-                {
-                    startButton.BackColor = System.Drawing.Color.LimeGreen;
-                    startButton.Enabled = true;
-                }
-                startRTGroupbox.BackColor = System.Drawing.Color.Gray;
-                loopBackBox.Enabled = false;
-                checkBox1.Enabled = false;
-
-                simulationSettingsGroupbox.BackColor = System.Drawing.Color.DarkGray;
-                comboMicrocontrollerBox.Enabled = false;
-                comboBox2.Enabled = false;
-                comboBox1.Enabled = false;
-                comboEncoderType.Enabled = false;
-                comboPLCType.Enabled = false;
-                LocalIPCombo.Enabled = false;
-
-                portGroupbox.BackColor = System.Drawing.Color.DarkGray;
-                ProdcheckBox.Enabled = false;
-                txtPLCIP.Enabled = false;
-                txtMcuCOMPort.Enabled = false;
-                txtWSCOMPort.Enabled = false;
-                txtPLCPort.Enabled = false;
+            // check each case (i.e. textbox) to make sure every value is valid
+            if (!MCUPortValid)
+            {
+                MessageBox.Show("MCU Port Invalid. Enter a valid port.");
             }
-            else if (finalSettings == true)
+            if (!PLCPortValid)
             {
-                // Editing Text to reflect state -- when finalized, you can click "edit settings"
-                acceptSettings.Text = "Finalize Settings";
-
-                startRTGroupbox.BackColor = System.Drawing.Color.DarkGray;
-                startButton.Enabled = false;
-                startButton.BackColor = System.Drawing.Color.LightGray;
-                loopBackBox.Enabled = true;
-                checkBox1.Enabled = true;
+                MessageBox.Show("PLC Port Invalid. Enter a valid port.");
+            }
+            if (!WCOMPortValid)
+            {
+                MessageBox.Show("WeatherCOM Port Invalid. Enter a valid port");
+            }
+            if (!MCUIPValid)
+            {
+                MessageBox.Show("Invalid MCU IP Address. Check your formatting and try again.");
```

Comment:
This is a good approach using `MessageBox.Show`, but it could be annoying to the user if they have to click through several Message Boxes. I recommend having the UI only display one message box that has each error string on a separate line.
For example, for each validator error that comes back false, you could have an error string accumulate each error message with `\n`